packages/tracking-sdk/src/proxy.ts

```diff
@@ -0,0 +1,21 @@
+export function readonly<T extends object, K extends keyof T>(
+  obj: T,
+  callback?: (key: K, value: T[K]) => void,
+): T {
+  return new Proxy(obj, {
+    get(target: T, prop) {
+      const val = target[prop as K];
+
+      if (val !== undefined) {
+        callback?.(prop as K, val);
+      }
+
+      return target[prop as K];
+    },
+    set(_target, prop, _value) {
+      throw new Error(
```

Comment:
We shouldn’t throw in other people code. This library already does it, but let’s not introduce new throws at least

Reply:
I don't know if silently ignoring the new value is better, to be honest. And if we allow them to set the value then we might get invalid feature flags in our events. It's better to explicitly disallow this behavior.